lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
@deeky666 have you done any tests -- are you just saying this? I've done tests to prove otherwise.

I just did another test from other servers man -- and I am able to use my ORIGINAL fix (using client-encoding), from a client using 8.4!